### ISSUE TITLE
Issues related to timeouts remain open forever

### ISSUE THREAD

**Issue by rouault** [google/oss-fuzz#645]
A timeout issue has been raised in https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=1599 and I committed a fix long time ago

Yet, the issue is still opened and when looking at the test case https://oss-fuzz.com/v2/testcase-detail/4613422639480832?noredirect=1 it appears it still reproduces it, but locally I can't reproduce it: it runs very fast.

I suspect the ogr_fuzzer program used might hit other timeout conditions, but the system doesn't distinguish them with the one it originally reports. If that's the case, that's annoying since I have no clue on how to fix those other issues. 

Same for https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=1644 / https://oss-fuzz.com/v2/testcase-detail/6267961146867712?noredirect=1

Actually when looking at https://oss-fuzz.com/v2/performance-report/libFuzzer_gdal_mitab_mif_fuzzer/libfuzzer_asan_gdal/latest , I can see reports related to timeouts, but how can I download the mentionned reproducer in the report ?

**Comment by inferno-chromium** [comment]
Unreproducible testcases (see Reproducible: NO in report) get auto-closed after a month, see "Closure:	Will be auto-closed on 6/27/2017 10:32:31 PM" on testcase report. Unreproducible testcases as the name says is a flaky test that reproduced only once during fuzzing. It is weird that this reproducible timeout turned into unreproducible. @Dor1s - can you check.

**Comment by Dor1s** [comment]
Sounds weird, looking.

**Comment by Dor1s** [comment]
Re [1599](https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=1599), I can reproduce it with the build downloaded from the [testcase page](https://oss-fuzz.com/v2/testcase-detail/4613422639480832?noredirect=1).

The same applies for [1644](https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=1644). I have a clue what might happen here, it can be OOM vs Timeout case, as peak_rss_mb goes to 1300 for 25 seconds, 1830 for 35, etc, but no clue for [issue 1599](https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=1599), as it doesn't use lot of memory.

**Comment by Dor1s** [comment]
I also verified that both minimized and non minimized testcases behave the same way.

**Comment by rouault** [comment]
The build referenced in https://oss-fuzz.com/v2/testcase-detail/4613422639480832?noredirect=1 is called https://storage.cloud.google.com/clusterfuzz-builds/gdal/gdal-undefined-201705141650.zip , so I guess it is the build on May 14th where the issue was first detected, but I pushed a fix on May 16th. With newer builds it is no longer reproducible, so why isn't this issue closed as fixed like others are ?

**Comment by oliverchang** [comment]
There are issues with consistently reproducing some OOMs/Timeouts (tracked at #314). These testcases will get marked as unreproducible, and we won't attempt to automatically detect when they're fixed.

**Comment by Dor1s** [comment]
@rouault it's an issue on our side, as @inferno-chromium said in https://github.com/google/oss-fuzz/issues/645#issuecomment-305632111 those testcases somehow got marked as Unreproducible. Due to that, verification of fixes is not being performed for them, and your fixes haven't been detected.

Sometimes we find really flaky crashes, so that Reproducible -> Unreproducible downgrade makes sense, but here it doesn't seem to be a correct thing...

**Comment by Dor1s** [comment]
I manually marked these two testcases as Fixed, since they actually are. Thus, the new timeouts (that you've seen in the performance report) should get filed soon.

@rouault, sorry for the inconveniences and thanks for your report.

I'll keep this issue open and assigned to myself for further investigation.

**Comment by oliverchang** [comment]
Note also that these bugs will automatically be closed as WontFix 30 days after they get marked as unreproducible.

**Comment by rouault** [comment]
Will marking the testcase as fixed will cause the ticket to be closed soon (ie within one day) ? Waiting for 30 days to get new tickets related to OOM/timout is a long time.

I already tried to marked the test cases as fixed in other tickets, but that doesn't seem to cause the ticket to be closed.

For example in that non-OOM/timeout situation, https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=1583 points to https://oss-fuzz.com/v2/testcase-detail/5709498763182080?noredirect=1 , which I marked as fixed a few days ago, but ticket 1583 is still opened. What is surprising is that https://oss-fuzz.com/v2/testcase-detail/5709498763182080?noredirect=1 points to ticket 1582... (which is well closed now). That's perhaps a different issue (1582 and 1583 apparently belongs to the same "group")

**Comment by oliverchang** [comment]
> Will marking the testcase as fixed will cause the ticket to be closed soon (ie within one day) ? 
Unfortunately, no. Marking testcases as Fixed manually on ClusterFuzz won't cause the monorail bug to be opened sooner. We will remove the ability to do so soon as it can be confusing. Thanks for bring that up!

> Waiting for 30 days to get new tickets related to OOM/timout is a long time.
You will continue to get tickets related to OOM/timeout if the existing one is marked as "unreproducible" -- right now we only allow report a single "Reproducible" testcase at any time. If it either gets properly detected as fixed, or becomes flaky, new tickets will still get filed when they're found.

**Comment by oliverchang** [comment]
The root cause of these issues is flaky timeouts, which are tracked in various other bugs. Let's close this for now.

To correct myself in an earlier comment, marking marking the testcase as "Fixed" on ClusterFuzz  usually *does* cause the monorail bug to be closed after a short while, as long as the bug association is correct, and there aren't any other ClusterFuzz testcases associated with the monorail bug.

For non-OOM/timeouts, https://oss-fuzz.com/v2/testcase-detail/5709498763182080?noredirect=1 Looks like there might have been a race condition somewhere with the grouping and filing. Will track this in a separate bug.
